File: benchmarks/tsb/bench_series_cumsum.ts
/**
 * Benchmark: series_cumsum — cumulative sum on 100k-element Series
 */
import { Series, cumsum } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 5;
const ITERATIONS = 20;

const data = Array.from({ length: ROWS }, (_, i) => i * 0.001);
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) {
  cumsum(s);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  cumsum(s);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "series_cumsum",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_series_cumsum.py
"""Benchmark: series_cumsum — cumulative sum on 100k-element Series"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 5
ITERATIONS = 20

data = np.arange(ROWS, dtype=np.float64) * 0.001
s = pd.Series(data)

for _ in range(WARMUP):
    s.cumsum()

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.cumsum()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "series_cumsum",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
